pytential/symbolic/pde/scalar.py

```diff
@@ -48,110 +66,157 @@ def __init__(self, kernel, use_l2_weighting):
         """
         return sym.var(name)
 
+    def representation(self, u, qbx_forced_limit=None, **kwargs):
+        """
+        :returns: a representation for the unknowns based on an integral
+            equation with density *u*. If *qbx_forced_limit* denotes an
+            on-surface evaluation, the corresponding jump relations are not
+            added to the representation.
+        """
+        raise NotImplementedError
+
+    def operator(self, u, **kwargs):
+        """
+        :returns: a boundary integral operator with corresponding jump
+            relations that can be used to solve for the density *u*.
+        """
+        raise NotImplementedError
+
 # }}}
 
 
 # {{{ dirichlet
 
 class DirichletOperator(L2WeightedPDEOperator):
     """IE operator and field representation for solving Dirichlet boundary
-    value problems with scalar kernels (e.g.
-    :class:`sumpy.kernel.LaplaceKernel`,
-    :class:`sumpy.kernel.HelmholtzKernel`, :class:`sumpy.kernel.YukawaKernel`)
+    value problems with scalar kernels (e.g. :class:`~sumpy.kernel.LaplaceKernel`,
+    :class:`~sumpy.kernel.HelmholtzKernel`, :class:`~sumpy.kernel.YukawaKernel`)
 
     .. note ::
 
         When testing this as a potential matcher, note that it can only
         access potentials that come from charge distributions having *no* net
-        charge. (This is true at least in 2D.)
+        charge. (This is true at least in 2D)
+
+    Inherits from :class:`L2WeightedPDEOperator`.
 
     .. automethod:: is_unique_only_up_to_constant
     .. automethod:: representation
     .. automethod:: operator
+    .. automethod:: __init__
     """
 
-    def __init__(self, kernel, loc_sign, alpha=None, use_l2_weighting=False,
-            kernel_arguments=None):
+    def __init__(self, kernel: Kernel, loc_sign: int, *,
+            alpha: Optional[Number] = None,
+            use_l2_weighting: bool = False,
+            kernel_arguments: Optional[Dict[str, Any]] = None):
         """
-        :arg loc_sign: +1 for exterior, -1 for interior
-        :arg alpha: the coefficient for the combined-field representation
+        :param loc_sign: :math:`+1` for exterior or :math:`-1` for interior
+            problems.
+        :param alpha: the coefficient for the combined-field representation
             Set to 0 for Laplace.
         """
         assert loc_sign in [-1, 1]
-
-        from sumpy.kernel import Kernel, LaplaceKernel
         assert isinstance(kernel, Kernel)
 
         if kernel_arguments is None:
             kernel_arguments = {}
-        self.kernel_arguments = kernel_arguments
-
-        self.loc_sign = loc_sign
-
-        L2WeightedPDEOperator.__init__(self, kernel, use_l2_weighting)
 
         if alpha is None:
-            if isinstance(self.kernel, LaplaceKernel):
+            from sumpy.kernel import LaplaceKernel
+            if isinstance(kernel, LaplaceKernel):
                 alpha = 0
             else:
                 alpha = 1j
```

Comment:
I was trying to use this for Yukawa and wasn't sure what `alpha` should be. Is this `1j` meant for Helmholtz?

Reply:
`alpha = 0` is the right choice for Yukawa. (That conditional should be fixed to be Helmholtz-only.) The point of `alpha` is the [Brakhage-Werner trick](https://andreask.cs.illinois.edu/cs598apk-f19/notes.pdf#page=270) to get rid of spurious resonances for Helmholtz.

Reply:
Probably also deserves a comment. :)

Reply:
Added some comments and flipped the if in d52cd68